pkg/controllers/infracluster/infracluster_controller.go

```diff
@@ -194,12 +193,17 @@ func (r *InfraClusterController) ensureInfraCluster(ctx context.Context, log log
 			return nil, fmt.Errorf("error ensuring GCPCluster: %w", err)
 		}
 	case configv1.AzurePlatformType:
-		azureCluster := &azurev1.AzureCluster{}
-		if err := r.Get(ctx, client.ObjectKey{Namespace: defaultCAPINamespace, Name: r.Infra.Status.InfrastructureName}, azureCluster); err != nil && !kerrors.IsNotFound(err) {
-			return nil, fmt.Errorf("error getting InfraCluster object: %w", err)
+		if r.Infra.Status.PlatformStatus.Azure.CloudName == configv1.AzureStackCloud {
+			log.Info("%s cloud environment for platform %s is not supported", "environment", configv1.AzureStackCloud, "platform", configv1.AzurePlatformType)
+			return nil, errUnsupportedPlatformEnvironment
```

Comment:
When this is returned to a higher level function, does it get unwrapped and handled or does this go into a backoff?

Since it's private, I'm expecting at the moment it goes into backoff?

Reply:
Not sure on this one, can take a look after I have some meetings, but if @damdo sees this first, do you know? :)

Reply:
If you were handling it, it would be somewhere around https://github.com/openshift/cluster-capi-operator/blob/714c1746d8a9a014a07dc0ffac0bf5bf98750692/pkg/controllers/infracluster/infracluster_controller.go#L105-L114, but I don't see anything there right now handling this sentinel error specifically.

You'll want an `errors.Is` somewhere around there and then handle this in the same way you're currently handling a `nil` infra cluster (which we should probably also move to this sentinel pattern)

Reply:
I've added handling for that error, sorry I don't get what you mean by moving the infra cluster to the same pattern, do you mean the infra cluster should be assigned a sentinel error if it's nil that we handle here?